Doc/library/functools.rst

```diff
@@ -330,20 +330,40 @@ The :mod:`functools` module defines the following functions:
 
 .. function:: partial(func, /, *args, **keywords)
 
-   Return a new :ref:`partial object<partial-objects>` which when called
+   Return a new :ref:`partial object <partial-objects>` which when called
    will behave like *func* called with the positional arguments *args*
    and keyword arguments *keywords*. If more arguments are supplied to the
    call, they are appended to *args*. If additional keyword arguments are
    supplied, they extend and override *keywords*.
    Roughly equivalent to::
 
+      Placeholder = object()
+
       def partial(func, /, *args, **keywords):
+          # Trim trailing placeholders and count how many remaining
```

Comment:
I suggest keeping the current rough equivalent and just adding some examples that use `Partial`. The full equivalent is too messy to be useful in the main docs. Also, we already link to the source code from the top of the documentation page.